src/schema/root_query_type.ts

```diff
@@ -847,15 +847,23 @@ const RootQuery = new GraphQLObjectType({
               },
             };
             const query: Filter<MetadataModel> = { $expr: testcaseName };
+            /*
+             * If `product_version` is specified, query for empty product as well and
+             * merge the results (with the product-specific values taking precedence).
+             */
             if (_.has(args, 'product_version')) {
-              query.product_version = product_version;
+              query.product_version = { $in: [null, product_version] };
             }
-            const docs = await col.find(query);
-            const mergedMetadata = _.mergeWith(
-              {},
-              ..._.map(docs, 'payload'),
-              customMerge,
-            );
+            const payloads = await col.aggregate([
+              { $match: query },
+              /*
+               * Prefer specific product version over general (empty), then lower priority
+               * over higher priority number.
+               */
+              { $sort: { product_version: 1, priority: -1 } },
+              { $replaceWith: '$payload' },
```

Comment:
@mgrabovsky Could you please add here some comments, so it is more easier to understand what `$replaceWith` does in this place.

Reply:
Added a comment here.